lib/src/otx/backend/native/models/base.py

```diff
@@ -884,6 +885,14 @@ def _dispatch_label_info(label_info: LabelInfoTypes) -> LabelInfo:
                 label_groups=[label_info],
                 label_ids=[str(i) for i in range(len(label_info))],
             )
+        if isinstance(label_info, NewLabelCategories):
+            # Handle LabelCategories objects from datumaro
+            labels = label_info.labels
+            return LabelInfo(
+                label_names=labels,
+                label_groups=[labels],
+                label_ids=[str(i) for i in range(len(labels))],
+            )
```

Comment:
OTX label_info is used in many places and I think that replacing it with NewLabelCategories might have unintended consequences. We can think about refactoring it but for now, I would prefer to avoid refactoring the label_info and focus on updating the dataset class. So I would suggest to do the conversion of NewlabelCategories into LabelInfo in the constructor of the new OTXDataset.

Later when we have finished migrating to the new dataset class, we can consider streamlining the code by removing the LabelInfo and also improve the OTXSample.

Reply:
Ok, thanks for the explanation. Let’s leave it like that.